scanner/k8s-assets/processor/processor.go

```diff
@@ -0,0 +1,179 @@
+		// The Service might already exist in the DB
+		// Let's try to fetch one Service by name
+		_serviceId, err := p.getService(ctx, serviceInfo)
+		if err != nil {
+			return "", fmt.Errorf("failed to create Service %s: %w", serviceInfo.Name, err)
+		}
+		serviceId = _serviceId
+	} else {
+		serviceId = createServiceResp.CreateService.Id
+	}
+	return serviceId, nil
+}
+
+// getService returns (if any) a ServiceID
+func (p *Processor) getService(ctx context.Context, serviceInfo scanner.ServiceInfo) (string, error) {
+	var serviceId string
+
+	listServicesFilter := client.ServiceFilter{
+		ServiceName: []string{serviceInfo.Name},
+	}
+	listServicesResp, err := client.ListServices(ctx, *p.Client, &listServicesFilter)
+	if err != nil {
+		return "", fmt.Errorf("Couldn't list services")
+	}
+
+	// Return the first item
+	if listServicesResp.Services.TotalCount > 0 {
+		for _, s := range listServicesResp.Services.Edges {
+			serviceId = s.Node.Id
+			break
+		}
+	} else {
+		return "", fmt.Errorf("ListServices returned no ServiceID")
+	}
+
+	return serviceId, nil
+}
+
+// ProcessPod processes a single pod and its containers
+func (p *Processor) ProcessPod(ctx context.Context, namespace string, serviceID string, podInfo scanner.PodInfo) error {
+	for _, containerInfo := range podInfo.Containers {
+		if err := p.ProcessContainer(ctx, namespace, serviceID, podInfo, containerInfo); err != nil {
+			return fmt.Errorf("failed to process container %s in pod %s: %w", containerInfo.Name, podInfo.Name, err)
+		}
+	}
+	return nil
+}
+
+func (p *Processor) getComponentVersion(ctx context.Context, manifest string) (string, error) {
+	var componentVersionId string
+
+	listComponentVersionFilter := client.ComponentVersionFilter{
+		Version: []string{manifest},
+	}
+	listCompoVersResp, err := client.ListComponentVersions(ctx, *p.Client, &listComponentVersionFilter)
+	if err != nil {
+		return "", fmt.Errorf("Couldn't list ComponentVersion")
+	}
+
+	if listCompoVersResp.ComponentVersions.TotalCount > 0 {
+		for _, cv := range listCompoVersResp.ComponentVersions.Edges {
+			componentVersionId = cv.Node.Id
+			break
+		}
+	} else {
+		return "", fmt.Errorf("ListComponentVersion returned no ComponentVersion objects")
+	}
+	return componentVersionId, nil
+}
+
+// ProcessContainer creates a ComponentVersion and ComponentInstance for a container
+func (p *Processor) ProcessContainer(ctx context.Context, namespace string, serviceID string, podInfo scanner.PodInfo, containerInfo scanner.ContainerInfo) error {
+
+	// Find component version by container image hash
+	componentVersionId, err := p.getComponentVersion(ctx, containerInfo.ImageHash)
+	if err != nil {
+		return fmt.Errorf("Couldn't find ComponentVersion")
+	}
+
+	// Create new CCRN
+	ccrn := CCRN{
+		Region:    p.config.RegionName,
+		Domain:    "x", // TODO
+		Project:   "x", // TODO
+		Cluster:   p.config.ClusterName,
+		Namespace: namespace,
+		Pod:       podInfo.Name,
+		PodID:     podInfo.Name, // Change this!
```

Comment:
we actually want the logical resource "Pod" and group by Pod. 

So probably the `ObjectMetadata.GenerateName` with the "-" at the end reduced is the correct name. 

The PodID should probably go away